### DISCUSSION TITLE
Purging permissions cache after applying a migration

### DISCUSSION THREAD

**Discussion by ohemelaar** [directus/directus#13019] [Archive]
Hello,

I'm deploying Directus on Kubernetes, and want to apply migrations automatically using a Job that spawns another Pod running Directus just to run the migration commands. I got all this working until I hit caching issues.

Indeed, I hoped I could run the migrations without restarting the Directus Pods that actually run the service. There's two reasons for this, the first one being to avoid downtime, but even if we had several replicas avoiding this also has the benefit to be less complex to set up. But then when caching is enabled for permissions I get `[FORBIDDEN]` errors when accessing collections. Indeed my migrations alter the permissions, but I don't see how I could invalidate the cache without restarting Directus. I tried tinkering with `CACHE_SYSTEM_TTL`, but I guess this only affects the API side and not the database cache. And I tested in memory or with Redis without success.

What would be the best way to handle such a situation?

**Comment by rijkvanzanten** [comment]
You could use the cache clear endpoint to reset the caches https://docs.directus.io/reference/system/utilities/#clear-the-internal-cache
